MantidQt/CustomInterfaces/test/ALCLatestFileFinderTest.h

```diff
@@ -16,36 +16,67 @@ using ScopedFileHelper::ScopedFile;
 using Mantid::Kernel::DateAndTime;
 
 /**
+ * Temporary directory that is deleted when it goes out of scope
+ */
+class ScopedDirectory {
+public:
+  /// Constructor: create directory in temp folder
+  ScopedDirectory(const std::string &dirName) : m_dirName(dirName) {
+    Poco::Path tmpPath(Mantid::Kernel::ConfigService::Instance().getTempDir());
+    tmpPath.pushDirectory(m_dirName);
+    m_directory = Poco::File(tmpPath);
+    m_directory.createDirectories();
+  }
+  /// Destructor: delete the directory
+  virtual ~ScopedDirectory() {
```

Comment:
Nothing, but I made the destructor virtual just in case somebody inherited from it in future. Perhaps it is better to make the class `final`.